solaris 10 chef.sh failed during `gem install`

I followed the instruction to create a solaris 10 box.
And it will always fail at chef.sh. Here is the error message.

```
Fetching: mixlib-config-2.1.0.gem (100%)
Fetching: mixlib-cli-1.5.0.gem (100%)
Fetching: mixlib-log-1.6.0.gem (100%)
Fetching: mixlib-authentication-1.3.0.gem (100%)
Fetching: mixlib-shellout-1.4.0.gem (100%)
Fetching: mime-types-1.25.1.gem (100%)
Fetching: systemu-2.6.4.gem (100%)
Fetching: ffi-1.9.3.gem (100%)
Building native extensions.  This could take a while...
ERROR:  Error installing chef:
        ERROR: Failed to build gem native extension.

        /opt/csw/bin/ruby18 extconf.rb
checking for ffi.h... no
checking for ffi.h in /usr/local/include,/usr/include/ffi... no
checking for rb_thread_blocking_region()... no
checking for rb_thread_call_with_gvl()... no
checking for rb_thread_call_without_gvl()... no
checking for ffi_prep_cif_var()... no
creating extconf.h
creating Makefile

make
Configuring libffi
make -C "/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi-i386-solaris2.10"
make[1]: Entering directory '/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi-i386-solaris2.10'
make "AR_FLAGS=" "CC_FOR_BUILD=" "CFLAGS=-Wall -fexceptions" "CXXFLAGS=" "CFLAGS_FOR_BUILD=" "CFLAGS_FOR_TARGET=" "INSTALL=/opt/csw/bin/ginstall -c" "INSTALL_DATA=/opt/csw/bin/ginstall -c -m 644" "INSTALL_PROGRAM=/opt/csw/bin/ginstall -c" "INSTALL_SCRIPT=/opt/csw/bin/ginstall -c" "JC1FLAGS=" "LDFLAGS=" "LIBCFLAGS=" "LIBCFLAGS_FOR_TARGET=" "MAKE=make" "MAKEINFO=/bin/bash /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/missing --run makeinfo " "PICFLAG=" "PICFLAG_FOR_TARGET=" "RUNTESTFLAGS=" "SHELL=/bin/bash" "exec_prefix=/usr/local" "infodir=/usr/local/share/info" "libdir=/usr/local/lib" "mandir=/usr/local/share/man" "prefix=/usr/local" "AR=false" "AS=as" "CC=gcc" "CXX=g++" "LD=/usr/ccs/bin/ld" "NM=/usr/ccs/bin/nm -p" "RANLIB=:" "DESTDIR=" all-recursive
make[2]: Entering directory '/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi-i386-solaris2.10'
Making all in include
make[3]: Entering directory '/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi-i386-solaris2.10/include'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi-i386-solaris2.10/include'
Making all in testsuite
make[3]: Entering directory '/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi-i386-solaris2.10/testsuite'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi-i386-solaris2.10/testsuite'
Making all in man
make[3]: Entering directory '/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi-i386-solaris2.10/man'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi-i386-solaris2.10/man'
make[3]: Entering directory '/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi-i386-solaris2.10'
restore=: && backupdir=".am$$" && \
am__cwd=`pwd` && CDPATH="${ZSH_VERSION+.}:" && cd /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi && \
rm -rf $backupdir && mkdir $backupdir && \
if (/bin/bash /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/missing --run makeinfo   --version) >/dev/null 2>&1; then \
  for f in /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/doc/libffi.info /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/doc/libffi.info-[0-9] /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/doc/libffi.info-[0-9][0-9] /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/doc/libffi.i[0-9] /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/doc/libffi.i[0-9][0-9]; do \
    if test -f $f; then mv $f $backupdir; restore=mv; else :; fi; \
  done; \
else :; fi && \
cd "$am__cwd"; \
if /bin/bash /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/missing --run makeinfo     -I doc -I /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/doc \
 -o /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/doc/libffi.info /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/doc/libffi.texi; \
then \
  rc=0; \
  CDPATH="${ZSH_VERSION+.}:" && cd /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi; \
else \
  rc=$?; \
  CDPATH="${ZSH_VERSION+.}:" && cd /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi && \
  $restore $backupdir/* `echo ".//opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/doc/libffi.info" | sed 's|[^/]*$||'`; \
fi; \
rm -rf $backupdir; exit $rc
/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/missing: line 52: makeinfo: command not found
WARNING: `makeinfo' is missing on your system.  You should only need it if
         you modified a `.texi' or `.texinfo' file, or any other file
         indirectly affecting the aspect of the manual.  The spurious
         call might also be the consequence of using a buggy `make' (AIX,
         DU, IRIX).  You might want to install the `Texinfo' package or
         the `GNU make' package.  Grab either from any GNU archive site.
/bin/bash ./libtool --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi  -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src  -g  -Wall -fexceptions -c -o src/debug.lo /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/debug.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src -g -Wall -fexceptions -c /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/debug.c  -fPIC -DPIC -o src/.libs/debug.o
/bin/bash ./libtool --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi  -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src  -g  -Wall -fexceptions -c -o src/prep_cif.lo /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/prep_cif.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src -g -Wall -fexceptions -c /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/prep_cif.c  -fPIC -DPIC -o src/.libs/prep_cif.o
/bin/bash ./libtool --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi  -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src  -g  -Wall -fexceptions -c -o src/types.lo /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/types.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src -g -Wall -fexceptions -c /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/types.c  -fPIC -DPIC -o src/.libs/types.o
/bin/bash ./libtool --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi  -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src  -g  -Wall -fexceptions -c -o src/raw_api.lo /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/raw_api.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src -g -Wall -fexceptions -c /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/raw_api.c  -fPIC -DPIC -o src/.libs/raw_api.o
/bin/bash ./libtool --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi  -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src  -g  -Wall -fexceptions -c -o src/java_raw_api.lo /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/java_raw_api.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src -g -Wall -fexceptions -c /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/java_raw_api.c  -fPIC -DPIC -o src/.libs/java_raw_api.o
/bin/bash ./libtool --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi  -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src  -g  -Wall -fexceptions -c -o src/closures.lo /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/closures.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src -g -Wall -fexceptions -c /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/closures.c  -fPIC -DPIC -o src/.libs/closures.o
/bin/bash ./libtool --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi  -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src  -g  -Wall -fexceptions -c -o src/x86/ffi64.lo /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/x86/ffi64.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src -g -Wall -fexceptions -c /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/x86/ffi64.c  -fPIC -DPIC -o src/x86/.libs/ffi64.o
/bin/bash ./libtool   --mode=compile gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi  -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src  -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src  -c -o src/x86/unix64.lo /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/x86/unix64.S
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src -c /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/x86/unix64.S  -fPIC -DPIC -o src/x86/.libs/unix64.o
/bin/bash ./libtool --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi  -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src  -g  -Wall -fexceptions -c -o src/x86/ffi.lo /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/x86/ffi.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src -g -Wall -fexceptions -c /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/x86/ffi.c  -fPIC -DPIC -o src/x86/.libs/ffi.o
/bin/bash ./libtool   --mode=compile gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi  -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src  -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src  -c -o src/x86/sysv.lo /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/x86/sysv.S
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src -I. -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/include -Iinclude -I/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src -c /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/src/x86/sysv.S  -fPIC -DPIC -o src/x86/.libs/sysv.o
/bin/bash ./libtool --tag=CC   --mode=link gcc -g  -Wall -fexceptions -version-info `grep -v '^#' /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi/libtool-version`    -o libffi.la -rpath /usr/local/lib src/debug.lo src/prep_cif.lo src/types.lo src/raw_api.lo src/java_raw_api.lo src/closures.lo                      src/x86/ffi64.lo src/x86/unix64.lo src/x86/ffi.lo src/x86/sysv.lo      
libtool: link: gcc -shared  -fPIC -DPIC -Wl,-z -Wl,text -Wl,-h -Wl,libffi.so.5 -o .libs/libffi.so.5.0.10  src/.libs/debug.o src/.libs/prep_cif.o src/.libs/types.o src/.libs/raw_api.o src/.libs/java_raw_api.o src/.libs/closures.o src/x86/.libs/ffi64.o src/x86/.libs/unix64.o src/x86/.libs/ffi.o src/x86/.libs/sysv.o     
libtool: link: (cd ".libs" && rm -f "libffi.so.5" && ln -s "libffi.so.5.0.10" "libffi.so.5")
libtool: link: (cd ".libs" && rm -f "libffi.so" && ln -s "libffi.so.5.0.10" "libffi.so")
libtool: link: ( cd ".libs" && rm -f "libffi.la" && ln -s "../libffi.la" "libffi.la" )
/bin/bash ./libtool --tag=CC   --mode=link gcc -g  -Wall -fexceptions   -o libffi_convenience.la  src/debug.lo src/prep_cif.lo src/types.lo src/raw_api.lo src/java_raw_api.lo src/closures.lo                      src/x86/ffi64.lo src/x86/unix64.lo src/x86/ffi.lo src/x86/sysv.lo      
libtool: link: rm -fr  .libs/libffi_convenience.a(D)
libtool: link: false cru .libs/libffi_convenience.a src/.libs/debug.o src/.libs/prep_cif.o src/.libs/types.o src/.libs/raw_api.o src/.libs/java_raw_api.o src/.libs/closures.o src/x86/.libs/ffi64.o src/x86/.libs/unix64.o src/x86/.libs/ffi.o src/x86/.libs/sysv.o 
Makefile:817: recipe for target 'libffi_convenience.la' failed
make[3]: *** [libffi_convenience.la] Error 1
make[3]: Leaving directory '/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi-i386-solaris2.10'
Makefile:1263: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory '/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi-i386-solaris2.10'
Makefile:509: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi-i386-solaris2.10'
libffi.mk:6: recipe for target '"/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi-i386-solaris2.10"/.libs/libffi_convenience.a' failed
make: *** ["/opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/libffi-i386-solaris2.10"/.libs/libffi_convenience.a] Error 2


Gem files will remain installed in /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3 for inspection.
Results logged to /opt/csw/lib/ruby/gems/1.8/gems/ffi-1.9.3/ext/ffi_c/gem_make.out
ERROR: exit code 1
Error executing command pfexec bash -l ./chef.sh : Exitcode was not what we expected
Exitcode was not what we expected
```

--- a/templates/solaris-10-ga-x86/README.md
+++ b/templates/solaris-10-ga-x86/README.md
@@ -22,17 +22,15 @@
    It also shuts down the box in the process, to allow packaging.
 
 3. Export the basebox and use it freely.
-   
-   To to this, from the definition folder execute (or provide the correct paths to the Vagrantfile.pkg):
-   
+
    ```
-   vagrant package --vagrantfile Vagrantfile.pkg --base 'vagrant-solaris10' --output 'vagrant-solaris10.box'
+   veewee vbox export 'vagrant-solaris10'
    ```
 
    You may find some issues if you run this command from your veewee installation, due to the relationship between vagrant and veewee.
    It's advisable to switch to the system's ruby (`rvm use system`) and running the command directly from the definition folder.
-   
-   
+
+
 #Migration to VMWare
 VirtualBox VMs can easily be migrated to VMWare but, in the case of Solaris 10 (at least), some differences regarding the use of IDE and SCSI render the migrated box unusable.
 

--- a/templates/solaris-10-ga-x86/chef.sh
+++ b/templates/solaris-10-ga-x86/chef.sh
@@ -1,10 +1,7 @@
 # Chef needs these
 /opt/csw/bin/pkgutil -y -i CSWgmake
 /opt/csw/bin/pkgutil -y -i CSWgcc4g++ CSWgcc4core
+/opt/csw/bin/pkgutil -y -i CSWcurl
 
-# These are needed to get a compiler working
-# Mainly because chef depends on compiling some native gems
-PATH=/opt/csw/bin:/opt/csw/gnu/:/opt/csw/gcc4/bin:$PATH
-export PATH
-
-/opt/csw/bin/gem install chef  --no-ri --no-rdoc
+# Install Chef
+curl -L https://www.opscode.com/chef/install.sh | bash

--- a/templates/solaris-10-ga-x86/definition.rb
+++ b/templates/solaris-10-ga-x86/definition.rb
@@ -1,8 +1,9 @@
 Veewee::Definition.declare({
   :cpu_count => '1', :memory_size=> '768',
   #Disk size needs to be 12Gig +
-  :disk_size => '65140', :disk_format => 'VDI', :hostiocache => 'off', :hwvirtex => 'on',
-  :os_type_id => 'Solaris_64',
+  :disk_size => '65140', :disk_format => 'VDI', :hostiocache => 'off',
+  :virtualbox => { :vm_options => [ "hwvirtex" => "on" ] },
+  :os_type_id => 'OpenSolaris_64',
   :iso_file => "sol-10-u11-ga-x86-dvd.iso",
   :iso_src => "",
   :iso_download_instructions => "- You need to download this manually as there is no automated way to do it\n"+
@@ -12,7 +13,7 @@ Veewee::Definition.declare({
     "- For other version: changed the iso filename+checksum\n",
   :iso_md5 => "aae1452bb3d56baa3dcb8866ce7e4a08",
   :iso_download_timeout => 1000,
-  :boot_wait => "10", :boot_cmd_sequence => [ 
+  :boot_wait => "10", :boot_cmd_sequence => [
     'e',
     'e',
     '<Backspace>'*22,

--- a/templates/solaris-10-ga-x86/postinstall.sh
+++ b/templates/solaris-10-ga-x86/postinstall.sh
@@ -10,6 +10,10 @@ PATH=/usr/bin:/usr/sbin:$PATH
 export PATH
 
 yes|/usr/sbin/pkgadd -d http://mirror.opencsw.org/opencsw/pkgutil-`uname -p`.pkg all
+
+# Uncomment this and pick a fast mirror from http://mirror.opencsw.org/status/
+# echo "mirror=http://www.grangefields.co.uk/mirrors/csw/testing" >> /etc/opt/csw/pkgutil.conf
+
 /opt/csw/bin/pkgutil -U
 
 # get 'sudo'
@@ -48,7 +52,7 @@ echo "LookupClientHostnames=no" >> /etc/ssh/sshd_config
 #
 ## no solaris2.11 .... mkheaders here ! needs some fixing ??
 ## /opt/csw/gcc4/libexec/gcc/i386-pc-solaris2.10/4.3.3/install-tools/mkheaders
-#/opt/csw/gcc4/libexec/gcc/i386-pc-solaris2.8/4.3.3/install-tools/mkheaders 
+#/opt/csw/gcc4/libexec/gcc/i386-pc-solaris2.8/4.3.3/install-tools/mkheaders
 #
 #/opt/csw/sbin/alternatives --display rbconfig18
 #/opt/csw/sbin/alternatives --set rbconfig18 /opt/csw/lib/ruby/1.8/i386-solaris2.9/rbconfig.rb.gcc4

